### ISSUE TITLE
Abandoned?

### ISSUE THREAD

**Issue by RomainLanz** [cgross/generator-cg-angular#112]
Hi @cgross,

I want to know if you have abandoned this project or not?

Cause you have answered to an issue 6 days ago but didn't commit, answer to PR, etc.

I think everyone here want an answer, and maybe you can redirect to this fork https://github.com/leanovate/generator-cc-angular if you don't want to continue to maintain this project.

**Comment by dancancro** [comment]
@RomainLanz [here](http://www.dancancro.com/compare-app-technologies/) is a detailed collection/comparison of options. If you know generator-cc-angular well, i warmly welcome you to do a questionnaire for it, by clicking [here](http://www.dancancro.com/technology-questionnaires/), so that its features can be included for everyone to see and compare. Choose Application Starters from the questionnaires list.

**Comment by martin-s** [comment]
In my opinion, there are no real benefits of other generator(s). If you want new packages, simply update your json-file and that's it.

**Comment by dancancro** [comment]
@martin-s here is the trade-off set between one of the other generators, Cleverstack, and this generator:

CleverStack offers these benefits that generator-cg-angular doesn't
* 1.8% Sponsoring company
* 1.3% Modularized Functionality
* 0.9% Setup persistent storage
* 0.9% Basic sourcecode organization, side->module(->submodule), with type subfolders
* 0.9% Approach to data readiness, prevents Flash of Unstyled/compiled Content (FOUC)
* 0.9% Module manager, using npm
* 0.9% Server-side integration & unit tests
* 0.9% Server-side performance tests
* 0.8% Generator framework, Command line interface (CLI)
* 0.7% Build tool / Task runner, using gulp
* 0.6% Code coverage reporting
* 0.5% Database connection pooling
* 0.5% Clustered HTTP sessions
* 0.4% Dedicated/searchable user group for questions, response time mostly under a day
* 0.4% Approach to AngularJS controllers, Controllers contain no XHR calls
* 0.4% Supports triggers
* 0.4% Supports transactions
* 0.4% Module manager
* 0.4% Dedicated/searchable user group for questions, using github issues
* 0.4% Approach to frontend code loading, Use AMD with Require.js
* 0.4% JavaScript Conventions, Don't use JQuery
* 0.4% End-to-end tests
* 0.4% Email user interface
* 0.3% Cross-Origin Resource Sharing (CORS)
* 0.3% Continuous integration (CI)
* 0.3% Command-line interface
* 0.3% Approach to error handling, System prevents users from seeing ugly errors
* 0.3% Pagination, paging (client-side)
* 0.3% Authentication, using Oauth
* 0.2% Database migration/evolution system
* 0.2% Starter Tested
* 0.2% Server-side performance monitoring & instrumentation
* 0.2% Safe from data tampering
* 0.2% Safe from cross-site request forgery CSRF
* 0.2% Authorization, Role-based
* 0.2% Authentication, Basic
* 0.2% Generate routes
* 0.2% Style guide for code
* 0.2% Approach to data readiness, Use a waiting image/spinner/progressbar
* 0.2% Share code between projects
* 0.2% Development build script, html page processing
* 0.2% back end generator
* 0.2% Account Management, register/login/logout
* 0.2% Authentication, External to the app, screencast of it, configured by route/state, using Oauth, using Passport, Multiple local Passport policies
* 0.2% Authentication, Cookie
* 0.2% Authentication, Google services
* 0.2% Authentication, Local
* 0.2% Object-relational mapping
* 0.2% Object-document mapping
* 0.2% Menus system
* 0.1% model (client side) generator
* 0.1% Read-eval-print-loop (Repl)
* 0.1% Serve URLs based on the route definitions
* 0.1% No pluralization, of table/collection names
* 0.1% No pluralization, of endpoint URLs
* 0.1% Production build script, concatenation (aggregation, globbing), minification, safe pre-minification, using gulp-ng-annotate
* 0.1% config generator
* 0.1% Browser detection
* 0.1% Documentation
* 0.1% Form validation
* 0.1% Responsive design
* 0.1% Authentication, Digest
* 0.1% Authentication, Token
* 0.1% Separate Prod/Dev DB config automation
* 0.1% Admin page for users and roles
* 0.1% Reversible data deletion
* 0.1% Development build script, check node version, run concurrent tasks in parallel to speed it up, reload build script file upon change, copy assets to build or dist or target folder, html page processing, change asset references to prevent caching
* 0.1% Production build script, generate build reports
* 0.1% Production build script, generate docs
* 0.1% controller (client side) generator
* 0.1% factory generator
* 0.1% front end generator
* 0.1% provider generator
* 0.1% Dedicated/searchable user group for questions, using Google Groups
* 0.1% Dedicated chatroom
* 0.1% Approach to data readiness, prevents Flash of Unstyled/compiled Content (FOUC), using ngCloak
* 0.1% Approach to frontend routing or state changing, example of it
* 0.1% Approach to XHR calls, using $resource (angular-resource)
* 0.1% Email message templates
* 0.1% Navigation bar
* 0.1% Sort - client side
* 0.1% Tables, dynamic (filtering and sorting)
* 0.1% Tables, responsive
* 0.1% Tables, export to CSV
* 0.1% Tables, using jqGrid (Guriddo)
* 0.1% Typeahead fields
* 0.1% Password encryption
* 0.1% https
* 0.1% Use a proxy to serve app
* 0.1% Approach to error handling, Server-side logging
* 0.1% Tables
* 0.1% Build timer
* 0.1% Auto Restart Server
* 0.1% module generator
* 0.1% test - client side
* 0.1% Choose client and/or server
* 0.1% Choose port numbers
* 0.1% Server setup options
* 0.1% controller (server side) generator
* 0.1% model (server side) generator
* 0.1% route (server side) generator
* 0.1% service (server side) generator
* 0.1% test (server side) generator
* 0.1% Alert messaging service
* 0.1% Chat
* 0.1% CSV processing
* 0.1% Currency list
* 0.1% E-mail sending system
* 0.1% E-mail sending system, using Sendgrid
* 0.1% E-mail sending system, using starter's own e-mail implementation
* 0.1% Stripe integration
* 0.1% Survey
* 0.1% Timezone list
* 0.1% Configurable response caching
* 0.1% Encrypted configuration
* 0.1% Serverside authenticated route restriction
* 0.1% Authorization, Access Contol List (ACL)
* 0.1% Separate runtime configuration profiles
* 0.1% Serve status codes only as responses
* 0.1% Accept nested, JSON parameters
* 0.1% Add a mutex around requests
* 0.1% Add timer header to requests
* 0.1% Support for signed and encrypted cookies
* 0.1% Serve headers only
* 0.0% Dedicated/searchable user group for questions
* 0.0% Dedicated/searchable user group for questions, using Stackoverflow
* 0.0% Source Maps
* 0.0% Development build script, reload build script file upon change
* 0.0% Development build script, Renames asset files to prevent browser caching
* 0.0% Development build script, check node version, run concurrent tasks in parallel to speed it up, reload build script file upon change, copy assets to build or dist or target folder, html page processing, Renames asset files to prevent browser caching, LESS/SASS/etc files are linted, compiled, Jade files compiled to HTML, html validation, JavaScript style checking, using Plato
* 0.0% Development build script, run end to end tests
* 0.0% Production build script, uglification
* 0.0% Production build script, compress images
* 0.0% Production build script, make static pages for SEO
* 0.0% Production build script, remove unused CSS
* 0.0% Account Management
* 0.0% Account Management, is password manager friendly
* 0.0% Account Management, Forgotten Password with Resetting
* 0.0% Account Management, Verify user registration by email
* 0.0% Account Management, Change password
* 0.0% Amazon workflow
* 0.0% Background task runner
* 0.0% Digital fingerprint
* 0.0% File Upload
* 0.0% Form generation
* 0.0% Menus system, submenus
* 0.0% Modals (popups)
* 0.0% SaaS
* 0.0% Shopping Cart
* 0.0% Starter Tested, test coverage is measured
* 0.0% Frontend Server/ Framework, using Node.JS, using Express
* 0.0% API Server/ Framework, using Restify
* 0.0% Setup persistent storage, using SQL db
* 0.0% Setup persistent storage, using SQL db, using MySQL
* 0.0% Setup persistent storage, using SQL db, using Postgresql
* 0.0% Setup persistent storage, using SQL db, using MariaDB
* 0.0% Setup persistent storage, using SQL db, using SQLite
* 0.0% Setup persistent storage, using SQL db, using SQL Server CE
* 0.0% Setup persistent storage, using NoSQL db
* 0.0% Setup persistent storage, using SQL db, using NoSQL db, using MongoDB
* 0.0% Approach to AngularJS views, Directives start with "data-"
* 0.0% Approach to AngularJS views, Dots in the model names
* 0.0% Approach to AngularJS directives, Only one directive definition per file
* 0.0% Approach to AngularJS controllers, Only one controller definition per file
* 0.0% Approach to AngularJS controllers, Don't load data/resolve in Controllers
* 0.0% Approach to AngularJS controllers, Only one view per controller
* 0.0% Approach to frontend routing or state changing, URLs start with '#!'
* 0.0% Approach to frontend routing or state changing, HTML5 Mode
* 0.0% Approach to safeguarding for minification, Use $inject for DI
* 0.0% Approach to safeguarding for minification, Use inline injection annotation for DI
* 0.0% Approach to error handling, Gracefully handles incompatible browsers
* 0.0% Approach to XHR calls, using $http and $q
* 0.0% Modularized Functionality, with submodules organized into packages
* 0.0% JavaScript (server side)
* 0.0% JavaScript Conventions, Don't use "for ... in"
* 0.0% JavaScript Conventions, Use 'use strict'
* 0.0% JavaScript Conventions, Don't use function expressions
* 0.0% Testing, using Chai
* 0.0% Testing, using Mocha
* 0.0% Testing, using Supertest
* 0.0% Stubbing and spying
* 0.0% Stubbing and spying, using Sinon
* 0.0% End-to-end tests, using Protractor
* 0.0% Continuous integration (CI), using Travis
* 0.0% Code coverage reporting, using Istanbul
* 0.0% Automated device testing, using Sauce Labs, using Browsersync, using Live Reload, using grunt
* 0.0% EditorConfig
* 0.0% Styled terminal output
* 0.0% Styled terminal output, using chalk
* 0.0% Development build script, check node version, run concurrent tasks in parallel to speed it up, reload build script file upon change, copy assets to build or dist or target folder, html page processing, update with data-* tags
* 0.0% Development build script, check node version, run concurrent tasks in parallel to speed it up, reload build script file upon change, copy assets to build or dist or target folder, html page processing, update with background imgs
* 0.0% Development build script, check node version, run concurrent tasks in parallel to speed it up, reload build script file upon change, copy assets to build or dist or target folder, html page processing, update with anchors images
* 0.0% Development build script, check node version, run concurrent tasks in parallel to speed it up, reload build script file upon change, copy assets to build or dist or target folder, html page processing, update with reference in input
* 0.0% Production build script, concatenation (aggregation, globbing), minification, of javascript
* 0.0% Production build script, concatenation (aggregation, globbing), minification, of css
* 0.0% Production build script, concatenation (aggregation, globbing), minification, safe pre-minification, dom munger, uglification, compress images, add CSS vendor prefixes, make static pages for SEO, cache templates, remove unused CSS, generate build reports, generate docs, using JSDoc
* 0.0% Command-line interface, using Independent command line interface
* 0.0% 404 Page
* 0.0% 500 Page
* 0.0% Account Management, recover password
* 0.0% Account Management, register/login/logout, recover password, is password manager friendly, Forgotten Password with Resetting, Remember me, Verify user registration by email, User profile images, Password strength meter, Change password, with double entry
* 0.0% Account Management, emails must be unique
* 0.0% Account Management, Admin can log in as other users
* 0.0% Calendar
* 0.0% Country list
* 0.0% Invoice
* 0.0% Menus system, collapsible
* 0.0% Menus system, sticky
* 0.0% Progress bars
* 0.0% Select box (fancy)
* 0.0% Select box (fancy), using Select2
* 0.0% Tables, using ngTable (doesn't need)
* 0.0% Tooltips
* 0.0% Job Queues
* 0.0% Authentication, External to the app
* 0.0% Authentication, External to the app, screencast of it, configured by route/state, using Oauth, Link multiple Oauth strategies to one account
* 0.0% Authentication, External to the app, screencast of it, configured by route/state, using Oauth, Oauth 1.0a
* 0.0% Authentication, External to the app, screencast of it, configured by route/state, using Oauth, Oauth 2.0
* 0.0% Authentication, External to the app, screencast of it, configured by route/state, using Oauth, Google strategy
* 0.0% Authentication, External to the app, screencast of it, configured by route/state, using Oauth, Facebook strategy
* 0.0% Authentication, External to the app, screencast of it, configured by route/state, using Oauth, Twitter strategy
* 0.0% Authentication, using Passport
* 0.0% Object-relational mapping, using Sequelize
* 0.0% Object-document mapping, using Mongoose


===================
===================


generator-cg-angular offers these benefits that CleverStack doesn't
* 2.6% Approach to frontend routing or state changing, State-based routing
* 0.9% Approach to frontend routing or state changing, example of it, screencast of it, State-based routing, using ui-router
* 0.4% Approach to data readiness, Don't use ng-controller
* 0.2% Bootstrap CSS, using UI Bootstrap AngularJS directives
* 0.1% view or view partial generator
* 0.1% Creates a favicon
* 0.0% Development build script, html templates put into $templateCache
* 0.0% Basic sourcecode organization, side->module(->submodule), without type subfolders
* 0.0% Approach to AngularJS modules, No global 'app' module variable
* 0.0% Approach to AngularJS modules, No global 'app' module variable without an IIFE
* 0.0% Testing, using Jasmine

**Comment by martin-s** [comment]
There are missing two important points: generator-cg-angular is a) simple and b) lightweight. I don't need tons of libraries to support authenticaton, database handling etc. Whenever I need such functionality, I can add it. Just my two cents :-)

**Comment by cgross** [comment]
Hi guys - this generator is definitely not abandoned.  I planned to do a big overhaul at some point soon but of course life always intervenes.  As @martin-s said, if you want newer version of some of the included libs, just download the updated versions using bower.  It would be nice if the generator had the most versions of the libs by default (and when I do the update it will) but updating them after your project is generated is trivial.

**Comment by eduardoinnorway** [comment]
I would like to participate in contributing for this generator, worked with it for over a year now and I love the concept, but there are still some things to be done :)

**Comment by magnis** [comment]
I love this for what it is. A mention of **overhaul** makes me a little nervous. I use the heck out of it as it does the job of quickly putting together an angular project, creating the base structure, linking the files for me and doing the final build. What the project does and how it does it, is out of scope. The only improvement that I think I could ask for it maybe make some of the already included things optional and make docs a little more clear on the best way to add what you want. (It took me a little bit to figure out how to add some things.)
